experimental/Schemes/CoherentSheaves.jl

```diff
@@ -622,7 +987,199 @@ function _pushforward(f::Hecke.Map{<:Ring, <:Ring}, I::Ideal, M::SubQuo)
+
+    # Assemble the restriction map from the parent node
+    if iszero(Asub)
+      # End of recursion. 
+      # Create a free module and the corresponding restriction morphism.
+      F = FreeMod(OO(V), ncols(Asub))
+      img_gens = elem_type(F)[]
+      for k in 1:j-1
+        push!(img_gens, F[k])
+      end
+      push!(img_gens, 
+            -sum([Ares[i, k]*F[(k>j ? k-1 : k)] for k in 1:ncols(Ares) if k!=j], 
+                 init=zero(F))
+           )
+      for k in j+1:ncols(Ares)
+        push!(img_gens, F[k-1])
+      end
+      res = hom(MU, F, img_gens, OOX(U, V))
+
+      # Since we are messing with the internals of the sheaf, we need 
+      # to leave everything clean. This includes manual caching.
+      add_incoming_restriction!(M, U, F, res)
+      object_cache(M)[V] = F
+      set_attribute!(F, :_presentation_matrix, Asub)
+      push!(return_patches, V)
+    else
+      # Intermediate recursion step. 
+      # Recreate the restriction of the module to the open subset but with one generator 
+      # less and construct the restriction map.
+      F, amb_res = change_base_ring(OOX(U, V), ambient_free_module(MU))
+      v = ambient_representatives_generators(MU)
+      M_gens = amb_res.(v)
+      rest_gens = [M_gens[k] for k in 1:length(M_gens) if k!=j]
+      rels = [amb_res(w) for w in relations(MU)]
+      MV = SubQuo(F, rest_gens, rels)
+      img_gens = elem_type(F)[]
+      for k in 1:j-1
+        push!(img_gens, M_gens[k])
+      end
+      push!(img_gens, 
+            -sum([Ares[i, k]*M_gens[k] for k in 1:length(M_gens) if k!=j], 
+                 init=zero(F))
+           )
+      for k in j+1:length(M_gens)
+        push!(img_gens, M_gens[k])
+      end
+      res = hom(MU, MV, MV.(img_gens), OOX(U, V))
+      add_incoming_restriction!(M, U, MV, res)
+      object_cache(M)[V] = MV
+      set_attribute!(MV, :_presentation_matrix, Asub)
+      return_patches = vcat(return_patches, _trivializing_covering(M, V))
+    end
+  end
+  return return_patches
+end
+
+@attr MatrixElem function _presentation_matrix(M::ModuleFP)
+  return matrix(map(presentation(M), 1))
+end
+
+function fill_with_lazy_glueings!(C::Covering, X::AbsCoveredScheme)
+  # TODO: Fill in all the implicit and restricted glueings from X.
+  # We assume that all patches in C are in a tree structure eventually 
+  # leading to affine charts of X.
+  return C
 end
 
+function _common_refinement_list(C::Covering, D::Covering)
+  patch_list = AbsSpec[]
+  for U in patches(C)
+    for V in patches(D)
+      success, W = _have_common_ancestor(U, V)
+      if success
+        #TODO: Model the intersection of both U and V with an appropriate 
+        #place in the tree structure.
+        incU = _flatten_open_subscheme(U, W)
+        incV = _flatten_open_subscheme(V, W)
+        UV = intersect(codomain(incU), codomain(incV))
+        push!(patch_list, UV)
+      end
+    end
+  end
+  return patch_list
+end
```

Comment:
please add a test for this function. And perhaps a few comments for docu?

Reply:
Looking at it now: This function is deprecated. I included its contents directly, so it will be removed.